src/Agent.Worker/Release/Artifacts/FileContainer/FileContainerItemCache.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Microsoft.TeamFoundation.Core.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Worker;
+using Microsoft.VisualStudio.Services.FileContainer;
+using Microsoft.VisualStudio.Services.FileContainer.Client;
+using Microsoft.VisualStudio.Services.WebApi;
+
+namespace Agent.Worker.Release.Artifacts.FileContainer
+{
+    public class FileContainerItemCache
+    {
+        public FileContainerItemCache(Uri teamProjectCollectionUrl, long containerId, string projectName, string accessToken, string rootItemPath = null, IExecutionContext logger = null, bool includeDownloadTickets = false)
```

Comment:
There is a blog pointer in AsyncLazy class, mainly we want to initialize the _items in a lazy fashion but we also want it to be asynchronous. Give that we are currently calling it only once it might be simpler to just create a async function, but for the time being I am keeping it as such.